src/aia_clic_extension.adoc

```diff
@@ -0,0 +1,1816 @@
+| 0x1401    |   31:0   | RW `clicintip[63:32]`     |   31:0    | RW `clicintie[63:32]`      | settings for interrupts 63 through 32
+6*^| ...
+| 0x147F    |   31:0   | RW `clicintip[4095:4064]` |   31:0    | RW `clicintie[4095:4064]`  | settings for interrupts 4095 through 4064
+|===
+
+
+== Same Privilege Mode Interrupt Preemption Support - smclic
+
+This extension requires smclicincr extension.
+
+The smclic extension extends interrupt preemption to support up to 256 interrupt
+levels for each privilege mode, where higher-numbered interrupt levels
+can preempt lower-numbered interrupt levels.  Interrupt level 0
+corresponds to regular execution outside of an interrupt handler.
+Levels 1--255 correspond to interrupt handler levels. Platform
+profiles will dictate how many interrupt levels must be supported.
+
+Incoming interrupts with a higher interrupt level can preempt an
+active interrupt handler running at a lower interrupt level in the
+same privilege mode, provided interrupts are globally enabled in this
+privilege mode.
+
+NOTE: Existing RISC-V interrupt behavior is retained, where incoming
+interrupts for a higher privilege mode can preempt an active interrupt
+handler running in a lower privilege mode, regardless of global
+interrupt enable in lower privilege mode.
+
+=== CLIC Interrupt Input Control (`clicintlvl`)
+
+`clicintlvl[__i__]` is an 8-bit WARL control register
+to specify interrupt levels 1-255. 
+0 is only valid when the interrupt is not implemented or visible.
+
+To select an interrupt to present to the core, the CLIC hardware
+combines the interrupt privilege mode encoding and
+`clicintlvl` to form an unsigned integer, then picks the global maximum
+across all pending-and-enabled interrupts based on this value.
+Next priority is determined using the AIA iprio rules. 
+Finally, the interrupt level of this selected interrupt is
+compared with the interrupt-level threshold of the associated privilege
+mode to determine whether it is qualified or masked by the threshold
+(and thus no interrupt is presented).
+
+NOTE: Implementations can choose to hardcode all iprio to the same non-zero value to simplify priority calculation.
+
+NOTE: Selecting an interrupt at a high privilege mode masks any
+interrupt at a lower privilege mode since the higher-privilege mode
+causes the interrupt signal to appear more urgent than any lower-privilege
+mode interrupt.
+
+NOTE: This register is defined as WARL as some implementations may want to support a limited number
+of values in this register including hardwiring some bits to fixed values.
+
+NOTE: Within a single privilege mode, it can be useful to separate interrupt
+handler tasks from application tasks to increase robustness, reduce
+space usage, and aid in system debugging.  Interrupt handler tasks
+have non-zero interrupt levels, while application tasks have an
+interrupt level of zero.
+
+=== New {mtvec} CSR Mode for CLIC

```

Comment:
mtvec is not really a preemption feature. As noted above, it should be its own extension.